.github/workflows/release-mockopampserver.yml

```diff
@@ -0,0 +1,94 @@
+# Release a tagged version of the '@elastic/mockopampserver' package.
+name: release-mockopampserver
+
+on:
+  push:
+    tags:
+      - mockopampserver-v*.*.*
+
+# 'id-token' perm needed for npm publishing with provenance (see
+# https://docs.npmjs.com/generating-provenance-statements#example-github-actions-workflow)
+permissions:
+  attestations: write
+  contents: write
+  packages: write
+  pull-requests: read
+  id-token: write
+
+jobs:
+  release:
+    runs-on: ubuntu-24.04
+    env:
+      PKGDIR: packages/mockopampserver
+      PKGNAME: "@elastic/mockopampserver"
+      DOCKER_IMAGE_NAME: ghcr.io/${{ github.repository }}/mockopampserver
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 'v22.16.0'
+          registry-url: 'https://registry.npmjs.org'
+
+      # Push a Docker image.
+      - uses: docker/setup-buildx-action@v3
+      - id: docker-meta
+        uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804  # v5.7.0
+        with:
+          images: ${{ env.DOCKER_IMAGE_NAME }}
+          flavor: |
+            latest=auto
+          tags: |
+            # git tag "mockopampserver-v1.2.3" -> Docker tags "1.2.3", "latest"
+            type=match,pattern=mockopampserver-v(\d+\.\d+\.\d+),group=1
+          labels: |
+            org.opencontainers.image.vendor=Elastic
+            org.opencontainers.image.title=mockopampserver
+            org.opencontainers.image.description=A mock OpAMP server, for dev and testing
+      - uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push Docker image
+        id: docker-push
+        uses: docker/build-push-action@1dc73863535b631f98b2378be8619f83b136f4a0  # v6.17.0
+        with:
+          context: ${{ env.PKGDIR }}
+          platforms: linux/amd64,linux/arm64
+          file: 'packages/mockopampserver/Dockerfile'
+          push: true
+          tags: ${{ steps.docker-meta.outputs.tags }}
+          labels: ${{ steps.docker-meta.outputs.labels }}
+      - name: Attest Docker image
+        uses: actions/attest-build-provenance@db473fddc028af60658334401dc6fa3ffd8669fd  # v2.3.0
```

Comment:
conversely there is no need to use the sha here, since it comes from the GH ecosystem, it's ok to use `@v2.3.0` or even just major `@v2`